### ISSUE TITLE
Create end point to return random quiz

### ISSUE THREAD

**Issue by silentkdev** [chingu-voyages/moonshot-chingu-quiz#74]
Front end has asked that we create an end point where a random quiz is returned from database.

**Comment by cMikolai** [comment]
Can we clarify a bit more what exactly we want? As far as I was thinking, when you visit the Quiz page where it has the filters on top, before you pick a filter, it would show quizzes below and we would like that these unfiltered quizzes are an overview of random ones? Or what were you guys thinking?

**Comment by silentkdev** [comment]
@Soreg @ZumDeWald

**Comment by Soreg** [comment]
Ok so basically what we (frontend) would like:
When we visit the quizzes page (overview), we call an endpoint to get random quizzes. These are the quizzes that we show when we haven't selected a subject or topic.
It _could_ be the same endpoint, where it just returns random quizzes if we didn't add a subject or query

**Comment by silentkdev** [comment]
@dpajek I think that the RANDOM function will work for now.  We'll always have opportunities to optimize and improve down the road.  I don't think there is any plans to let users upload their own quizzes at the moment so we don't have to worry about a run away train on the number of quizzes that we have to sort through.

If this is something you'd like to implement I can assign you the issue although I would ask you to wait working on it until I push out the refactor to the backend I am currently working on.

**Comment by dpajek** [comment]
@silentkdev sounds good! Yes please assign to me and let me know when the refactor is done.

**Comment by silentkdev** [comment]
@dpajek refactor is currently up for PR.  Hopefully will merge into dev tomorrow some time.  I've also assigned this issue to you.  I will help get you up an running once dev is ready.

**Comment by dpajek** [comment]
@silentkdev looks like the PR has been merged. Am I good to sync the dev branch? After syncing, it sounds like I just need to run the update schema script to make sure docker is updated.

**Comment by silentkdev** [comment]
@dpajek The PR has been merged.  You can pull the upstream dev branch and then branch off into your feature branch from there.  You will actually need to run the `reset-db` npm script.  You need to make sure you're docker containers are not currently running before running the script.  Afterwards, you can run the `npm run dev` script in the root folder if you're working on the entire project or just `docker-compose up` if you're only going to work on the api.  Let me know if you run into any issues.

I'm not sure how familiar you are with the back end but to get you started:

1.  We need a route for `/api/v1/quiz/random`.  You can add the route in the `routes/apiV1/quiz.js` file.  You'll see a number of routes already attached to that router.  This one will need to be a GET route.
2. The second parameter to the `router.get()` function is the controller.  Those can be found in the `controllers` directory.  You'll want to add the actual logic of getting the random quiz to a file in the `/controllers/apiV1/quiz` directory.  A file name along the lines of "get random quiz" will suffice.
3.  The postgres query for selecting a random quiz should be added to the `db/queries/quiz/select.js` file where the other select queries are to be found.

We only have one quiz so don't be alarmed when you keep getting the same quiz back testing the end point.  You'll also want to place the random route before the  `getById` route in `routes/apiV1/quiz.js` or else postgres will complain that the string "random" is not a properly formatted uuid

**Comment by dpajek** [comment]
@silentkdev Thanks for your instructions! I made a change to query for x random quizzes at quiz/random/## Let me know if you think we should handle it differently -- it sounded like @Soreg wanted to be able to query a bunch of random quizzes.

I wasn't really able to test it effectively because there's only one quiz. Is it easy to add dummy data? I couldn't figure out how to do it with postman (do I need to post the quiz and question data separately?) 

Anyway, let me know what you think. Happy to make any changes.
